compiler/wasm/src/compile.rs

```diff
@@ -85,32 +113,36 @@ pub fn compile(
     }
 }
 
-fn add_noir_lib(context: &mut Context, library_name: &str) {
-    let path_to_lib = Path::new(&library_name).join("lib.nr");
-    let library_crate_id = prepare_dependency(context, &path_to_lib);
-
-    add_dep(context, *context.root_crate_id(), library_crate_id, library_name.parse().unwrap());
-
-    // TODO: Remove this code that attaches every crate to every other crate as a dependency
-    let root_crate_id = context.root_crate_id();
-    let stdlib_crate_id = context.stdlib_crate_id();
-    let other_crate_ids: Vec<_> = context
-        .crate_graph
-        .iter_keys()
-        .filter(|crate_id| {
-            // We don't want to attach this crate to itself or stdlib, nor re-attach it to the root crate
-            crate_id != &library_crate_id
-                && crate_id != root_crate_id
-                && crate_id != stdlib_crate_id
-        })
-        .collect();
+fn process_dependency_graph(context: &mut Context, dependency_graph: DependencyGraph) {
+    let mut crate_names: HashMap<&CrateName, CrateId> = HashMap::new();
+
+    for lib in dependency_graph.root_dependencies {
+        let crate_id = add_noir_lib(context, lib);
+        crate_names.insert(lib, crate_id);
 
-    for crate_id in other_crate_ids {
-        context
-            .crate_graph
-            .add_dep(crate_id, library_name.parse().unwrap(), library_crate_id)
-            .unwrap_or_else(|_| panic!("ICE: Cyclic error triggered by {library_name} library"));
+        add_dep(context, *context.root_crate_id(), crate_id, lib.clone());
```

Comment:
```suggestion
        add_dep(context, *context.root_crate_id(), crate_id, lib);
```
We haven't removed the clone after https://github.com/noir-lang/noir/pull/3213#discussion_r1366795919

Reply:
I don't think I can retain a reference to `&lib` if `add_dep` takes ownership of it.

Maybe it would be best to refactor `CrateGraph` and save the crates' name when I register the lib? Then we could have `context.crate_id(&crate_name)` similar to `root_crate_id()`. WDYT?